Api/Configuration.md

```diff
@@ -0,0 +1,73 @@
+Configuration is the set of external parameters provided to an application that controls the application’s behaviour in some way. It typically consists of a mixture of settings and secrets that the application will load at runtime.
+
+The configuration system in ASP.NET Core is very flexible, allowing you to load configuration from a wide range of locations. Also, ASP.NET Core provides the ability to easily bind this configuration to strongly typed options objects, which then you can inject into your services. This model of strongly typed configuration makes it easy to logically group settings around a given feature and lends itself well to unit testing.
+
+The ASP.NET Core configuration model centers on two main constructs: *`ConfigurationBuilder`* and *`IConfiguration`*. *`ConfigurationBuilder`* describes how to construct the final configuration representation for your app, and *`IConfiguration`* holds the configuration values themselves.
+
+
+
+### Configuration providers
+
+ASP.NET Core uses configuration providers to load key-value pairs from a variety of sources. Applications can use many different configuration providers. You can load values from files, environment variables, command-line arguments, from a database, from a remote service, or you could create your own custom configuration provider.
+
+We use the *`CreateDefaultBuilder`* helper method to bootstrap *`HostBuilder`* for our app. The *`CreateDefaultBuilder`* method calls *`ConfigureAppConfiguration`* and sets up a number of default configuration providers.
+
+
+
+### Options pattern
+
+ASP.NET Core promotes the use of strongly typed settings as a way of letting configuration code adhere to the single responsibility principle and to allow the injection of configuration classes as explicit dependencies. Such settings also make testing easier. Instead of having to create an instance of *`IConfiguration`* to test a service, you can create an instance of the POCO options class.
+
+Your options classes need to be non-abstract and have a public parameterless constructor to be eligible for binding. The binder will set any public properties that match configuration values!
+
+To help facilitate the binding of configuration values to your custom options classes, ASP.NET Core introduces the *`IOptions<T>`* interface. This is a simple interface with a single property, Value, which contains your configured options class at runtime. Options classes are set up in the *`ConfigureServices`* section of *`Startup`*:
+
```

Comment:
_which contains your configured options class at runtime_
Class or instance ? :)
_Options classes are set_
Is Options setting are configured in the..